src/OpenRiaServices.Server.EntityFrameworkCore/Framework/EFCoreTypeDescriptor.cs

```diff
@@ -245,8 +245,18 @@ protected override IEnumerable<Attribute> GetMemberAttributes(PropertyDescriptor
 #else
                 bool isManyToMany = navigation.IsCollection && navigation.Inverse?.IsCollection == true;
 #endif
+
+#if NETSTANDARD2_0
                 if (!isManyToMany)
                 {
+#else
+                if (!isManyToMany
+                    // Don't generate association attributes for Owned types (onless they have all FK fields explictly defined, in which case they can be treated as Entities)
+                    //  if we generate association attributes then it cannot be treated as a ComplexObject
+                    && !(navigation.TargetEntityType.IsOwned() && navigation.ForeignKey.Properties.Any(static p => p.IsShadowProperty())))
```

Comment:
I believe the IsOwned check is actually redundant, 

```suggestion
                    && !(navigation.ForeignKey.Properties.Any(static p => p.IsShadowProperty())))
```

Reply:
I did the change but I am not 100% sure what we want to do for owned entities that have explicit keys.

- You can look at the test baseline how the new code is generated.
- It works well for owned entities which do not have a key